zerver/lib/addressee.py

```diff
@@ -111,17 +117,22 @@ def legacy_build(sender: UserProfile,
                 raise JsonableError(_("Cannot send to multiple streams"))
 
             if message_to:
-                stream_name = cast(str, message_to[0])
+                stream = message_to[0]
             else:
                 # This is a hack to deal with the fact that we still support
                 # default streams (and the None will be converted later in the
                 # callpath).
                 if sender.default_sending_stream:
                     # Use the users default stream
-                    stream_name = sender.default_sending_stream.name
+                    stream = sender.default_sending_stream.name
```

Comment:
Can you add a commit on top of this making this code path not go back through stream name, given that we're starting with a stream object?

Reply:
Sorry, but I am not sure what you mean. We kinda have to use the stream name here because the caller relies on the name to call `get_stream`. Also, `stream` isn't a `Stream` object, it can be either an ID or a stream name, in this case (maybe `stream_name_or_id` might be a better variable name, I'll change it to that). Or am I missing something here? :)

Reply:
`sender.default_sending_stream` is a Stream object.  So if we now support in the following code having a stream ID or a stream name, we could do `.id` here, not `.name`.